src/SwedbankPay.Sdk.Infrastructure/PaymentInstruments/Card/AddressDto.cs

```diff
@@ -0,0 +1,52 @@
+﻿using SwedbankPay.Sdk.PaymentOrders;
+
+namespace SwedbankPay.Sdk.PaymentInstruments.Card
+{
+    internal class AddressDto
+    {
+        public AddressDto() { }
+
+        public AddressDto(Address address)
+        {
+            City = address.City;
+            CoAddress = address.CoAddress;
+            CountryCode = address.CountryCode.Name;
+            Email = address.Email.ToString();
+            FirstName = address.FirstName;
+            LastName = address.LastName;
+            Msisdn = address.Msisdn.ToString();
+            StreetAddress = address.StreetAddress;
+            ZipCode = address.ZipCode;
+        }
+
+        public AddressDto(PickUpAddress address)
+        {
+            City = address.City;
+            CoAddress = address.CoAddress;
+            CountryCode = address.CountryCode.Name;
+            StreetAddress = address.StreetAddress;
+            ZipCode = address.ZipCode;
+            Name = address.Name;
+        }

```

Comment:
> 
> 
> This is starting to look very close to done! There are a few common patterns I would like to address:
> 
>     1. Don't take `List<>` as constructor arguments. Instead, initialize the `List<>` property in the constructor and allow consumers to `.Add()` items to it.

I've done this, but I think its not a great idea, as several places has these as required argument, and not passing them in the constructor makes it more error prone.

Reply:
Ok, I understand what you mean wrt. `hostUrls` and `prices`. Let's compromise and take them as required arguments, but only in singular form, so we initialize the lists in the constructor and keep the list properties readonly, but add the required items through the constructor?

Reply:
List items as singular form? I do not follow here.

Reply:
What I meant was that instead of taking `List<Price> prices`, take a `Price price` object in the constructor, and add that to the list (which is initialized in the constructor).

Reply:
You'll have to mock me up an example here because I do not follow.
From what I read then a `Price` would itself be a list of sub-`price` objects making this needlessly complex.
Or do you only want the possibility to add one `Price` through the constructor, so if a user want to add more than one `Price` they have to add the rest through the property instead of just passing it through the constructor?

Reply:
> Or do you only want the possibility to add one `Price` through the constructor, so if a user want to add more than one `Price` they have to add the rest through the property instead of just passing it through the constructor?

Yes!

Reply:
This sounds really backwards.

Reply:
Ok, then make the constructor take `IEnumerable<Price>` so it's possible to send `new[] { new Price(...) }` in the constructor and not requiring the consumer to construct an entire `List<Price>`. Does that sound like an OK compromise?

Reply:
I don't understand what you are trying to achieve here.

Reply:
`new[] { new Price(...) }` is simpler than `new List<Price>(new[] { new Price(...) })`.